src/shared/defaultAwsContextCommands.ts

```diff
@@ -170,17 +170,26 @@ export class DefaultAWSContextCommands {
     }
 
     /**
-     * @description Responsible for getting a profile from the user,
-     * working with them to define one if necessary.
+     * @description Responsible for getting a credential source from the user,
+     * working with them to create one if necessary.
      *
-     * @returns User's selected Profile name, or undefined if none was selected.
+     * @returns User's selected credential source, or undefined if none was selected.
      * undefined is also returned if we leave the user in a state where they are
      * editing their credentials file.
      */
-    private async getProfileNameFromUser(): Promise<string | undefined> {
+    private async getCredentialSourceFromUser(): Promise<string | undefined> {
         const credentialsFiles: string[] = await UserCredentialsUtils.findExistingCredentialsFilenames()
+        const providerMap = await CredentialsProviderManager.getInstance().getCredentialProviderNames()
+        const profileNames = Object.keys(providerMap)
 
-        if (credentialsFiles.length === 0) {
+        if (profileNames.length > 0) {
+            // There are credentials for the user to choose from
+            const dataProvider = new DefaultCredentialSelectionDataProvider(profileNames, ext.context)
+            const state = await credentialSourceSelector(dataProvider)
+            if (state && state.credentialProfile) {
```

Comment:
The name `state.credentialProfile` contradicts the rename of `credentialProfileSelector` to `credentialSourceSelector`.